### ISSUE TITLE
Replace Images? could this be an added Feature?

### ISSUE THREAD

**Issue by alandoyle95** [dtjohnson/xlsx-populate#64]
I have tried to use some ZIP lib's to just flat out replace an image in the excel file, but i think that this would be a pretty easy feature to add since you already have the Excel file extracted, all that would have to be added was what image do you want to replace and what you want to replace it with, and i think it would be really sweet if you can optionally keep all original styles (width height posX posY) and all that from the original file, but can override if you want too.

love your work, and excited to see it continue to expand, 

side note.. does this project take donations? if so where? :)

**Comment by dtjohnson** [comment]
Yeah, proper image support would be great. In the meantime, you could hack this yourself. The extracted JSZip in on the workbook as `_zip`.

I happily take donations of code. :)

**Comment by alandoyle95** [comment]
oo good to know, I shall try that, and I will get into the code and see how much of it i can pick up so can contribute

**Comment by alandoyle95** [comment]
the `drawing1.xml` file is going to be a fun one to crawl through. lol...

**Comment by dtjohnson** [comment]
Yeah, that's a tough one. There are also the embedded image files somewhere in the zip structure. We'd need to have some Drawing class for sure. It's tough to say the full object model from one example. You'd need to look at the XML for a number of different drawings or try to find some MS documentation.

**Comment by alandoyle95** [comment]
Yea the files are

/media/image1.png

/media/image2.jpg

On Aug 5, 2017 4:35 AM, "Dave T. Johnson" <notifications@github.com> wrote:

Yeah, that's a tough one. There are also the embedded image files somewhere
in the zip structure. We'd need to have some Drawing class for sure. It's
tough to say the full object model from one example. You'd need to look at
the XML for a number of different drawings or try to find some MS
documentation.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/dtjohnson/xlsx-populate/issues/64#issuecomment-320438766>,
or mute the thread

**Comment by alandoyle95** [comment]
well. hacked the crap out of this...
but I have a working way to swap images with image name inside of excel as well.
its a HORRID way of doing it. but gets the job done.. and seeing how I have never worked with jszip I don't think I did to horrible...

https://github.com/alandoyle95/xlsx-populate/commit/5d0f4a02e79612dd3945857bf374888052fd08cc

not worthy of a pull request though.... lol not even close...

**Comment by alandoyle95** [comment]
https://github.com/alandoyle95/xlsx-populate

My Fork has working image swapping, naming, getting of position (could be able to set if I knew anything about the unit of measurement) also works with relationships...

still needs more support for oneCellAcher images and such.. and I will be working on that...

needs some more refining, but I have used this code on 323 forms that are complex as they come that my company uses and it does not break charts, drawings, icons, shapes, so far its been doing very well...

would love some feedback and maybe some help implementing a system to allow adding new drawings with the position working.

still, need to add in tests and more documentation for this, but its a super ruff still so I am holding off on that until we have a better code layout.

any feedback from everyone would be awesome as I would love to see this integrated into the repo down the road as no other Lib has image manipulation that works as smooth as this one.

@dtjohnson

**Comment by waterfoul** [comment]
@alandoyle95 Could you update your fork with the latest code? I'd like to take a swing at some of the remaining pieces

**Comment by alandoyle95** [comment]
Yea I'm actually working on doing that as my fork is broken for the
transform function but the newest is not, and that would be amazing Haha,
the part I'm having issues with the excess unit of measurement..

**Comment by alandoyle95** [comment]
> @alandoyle95 Could you update your fork with the latest code? I'd like to take a swing at some of the remaining pieces

updated mine finally.. still doing testing, you are welcome to test as well lol

**Comment by dnt294** [comment]
@alandoyle95 Long times has passed ... Have you update your fork with latest code? 👀 🙏
